src/python/interpret/ext/extension_utils.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2019 Microsoft Corporation
+# Distributed under the MIT software license
+
+import logging
+import pkg_resources
+import re
+
+module_logger = logging.getLogger(__name__)
+
+blackbox_key = "interpret_ext_blackbox"
+
+
+def _validate_class_name(proposed_class_name):
+    """
+    Used to validate class names before registration.
+
+    :param proposed_class_name: The string to name the class
+    :type proposed_class_name: str
+    """
+    # regex for class name came from
+    # https://stackoverflow.com/questions/10120295
+    # TODO: Support other languages
+    match = re.match(r"[a-zA-Z_][a-zA-Z_0-9]+", proposed_class_name)
```

Comment:
Any need to impose a max length for any part of the naming of this stuff? I guess it only makes life worse for the person who's writing the extension, so probably no need

Reply:
Currently no, and it will fail on creation if python has an internal limit